ios-cr-task/cr.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Netguru iOS code review task
+//
+
+class paymentViewController: UIViewController {
+    var delegate: PaymentViewControllerDelegate
+    let customView = PaymentView()
+    let payment: Payment?
+
+    func callbacks() {
+        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
+        customView.didTapButton = {
+            if let payment = self.payment {
+                self.delegate.didFinishFlow(amount: payment.amount
+                                            currency: payment.currency)
+            }
+        }
+    }
+
+    func viewDidLoad() {
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        customView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        customView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        view.addSubview(customView)
+        fetchPayment()
+        setupCallbacks()
+    }
+
+    internal let ViewModel: PaymentViewModel
+
+    func fetchPayment() {
+        customView.statusText = "Fetching data"
+        ApiClient.sharedInstance().fetchPayment { payment in
```

Comment:
`sharedInstance()` is more preferable  name in objective c, in swift it would be better to use just `shared`.
By the way, having this kind of global ApiClient implemented with singleton can be considered as SIngle Responsibility violation. For servicing, i can suggest having service base layer, which will be provided in viewModel as inputs. 

In ViewModel you can have something like this
```suggestion
        struct PaymentViewModelInputs {
             let paymentService: PaymentServicing
        }
        
        final class PaymentViewModel {
              let inputs: PaymentViewModelInputs
              var payment: PublishSubject<Payment?> = PublishSubject()
              
              init(inputs: PaymentViewModelInputs) {
                   self.inputs = inputs
              }
              
              func fetchPayment() {
                  inputs.paymentService.fetch() { [weak self] payment in
                        self?.payment.onNext(payment)
                  }
              }
        }
```
As i mentioned above it will be better to have generic base view model with imputs.